dbt/include/global_project/macros/materializations/snapshots/helpers.sql

```diff
@@ -168,14 +168,25 @@
 {% endmacro %}
 
 
-{% macro build_snapshot_staging_table(strategy, sql, target_relation) %}
```

Comment:
from what I can tell no 1P adapters override this macro specifically but it is technically a breaking change and could cause issues in projects / adapters that have overridden it. 

I can think of a couple of less-satisfying implementation approaches that would make this change backward-compatible: 
1. pass an optional `snapshot_select_sql` to `build_snapshot_staging_table` and use it instead of calling `snapshot_staging_table` in this macro if availab.e 
2. make a redundant call to  `snapshot_staging_table` [above](https://github.com/dbt-labs/dbt-adapters/pull/243/files#diff-678f35d7828b12f8a6fb7f70c01683aadf4f3a1141b1b3d4ab89aa356938c2baR40-R41), just to get `dbt_updated_at_data_type`.
  * I don't think this should have real performance implications since its just a templating macro
  * It's less slick but does keep the macros more independent from one another's results / execution order, so I think I prefer this option slightly over (1)

What do you think?

Reply:
I've changed this to keep the original macro signature so we're getting the sql twice.